### PR TITLE
Add UFO dir argument validity checks and informative error messages

### LINKED ISSUE #1: Improve invalid source path error messages
```
$ target/release/ufofmt 1 2 3 4 5
[ERROR] Read error in "4": Io(Os { code: 2, kind: NotFound, message: "No such file or directory" })
[ERROR] Read error in "5": Io(Os { code: 2, kind: NotFound, message: "No such file or directory" })
[ERROR] Read error in "3": Io(Os { code: 2, kind: NotFound, message: "No such file or directory" })
[ERROR] Read error in "1": Io(Os { code: 2, kind: NotFound, message: "No such file or directory" })
[ERROR] Read error in "2": Io(Os { code: 2, kind: NotFound, message: "No such file or directory" })
```

### PR DESCRIPTION
**Pull request by chrissimpkins**
Closes #1

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -15,12 +15,33 @@ struct Opt {
...
+    // UFO dir validity check
+    // ~~~~~~~~~~~~~~~~~~~~~~
+    let mut bad_ufo_path = false;
+    for ufo_testpath in &argv.ufopaths {
+        match ufo_testpath.exists() {
+            true => (),
+            false => {
+                let error_str = "[ERROR]".red().bold();
+                eprintln!("{} {:?} is not a valid UFO directory path", error_str, ufo_testpath);
+                bad_ufo_path = true;
```
Comment (madig) [inline src/main.rs:34]: You could also exit right here.

Comment (chrissimpkins) [reply src/main.rs:34]: Trying to capture all invalid paths if there are > 1 UFO source file args.  Know of a better approach?

Comment (chrissimpkins) [reply src/main.rs:34]: e.g.,  with five bad ufo paths

```
target/release/ufofmt 1 2 3 4 5
[ERROR] "1" is not a valid UFO directory path
[ERROR] "2" is not a valid UFO directory path
[ERROR] "3" is not a valid UFO directory path
[ERROR] "4" is not a valid UFO directory path
[ERROR] "5" is not a valid UFO directory path
```

Comment (madig) [reply src/main.rs:34]: Ah, right. One could probably do a https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.filter and collect all the bad paths for display and immediate exit.

Comment (chrissimpkins) [reply src/main.rs:34]: > Ah, right. One could probably do a https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.filter and collect all the bad paths for display and immediate exit.

I like it!  I'll give it a try

Comment (chrissimpkins) [reply src/main.rs:34]: You should have full maintainer access btw.  Please feel free to add any changes that you'd like to see in a formatter client frontend for the lib.

Comment (madig) [reply src/main.rs:34]: Thanks! I may be silent for a while because there is so much stuff to do elsewhere, though.

Comment (chrissimpkins) [reply src/main.rs:34]: No worries at all!  Don't feel compelled to do anything.  Just making sure that you know that you are always welcome to contribute here.

Comment (chrissimpkins) [reply src/main.rs:34]: > Ah, right. One could probably do a https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.filter and collect all the bad paths for display and immediate exit.

466707a7bbe4e10b109c3885aa8c45b9c581615f

**Pull request merged**